src/UseCases/ValidateMembershipFee/ValidateFeeRequest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\MembershipContext\UseCases\ValidateMembershipFee;
+
+/**
+ * @license GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+class ValidateFeeRequest {
+
+	public const PERON_APPLICANT = 'person';
+	public const COMPANY_APPLICANT = 'firma';
+
+	private $membershipFee;
+	private $paymentIntervalInMonths;
+	private $applicantType;
+
+	public static function newInstance(): self {
```

Comment:
The fluent interface and builder pattern looks nice, because it is a good substitute for named constructor parameters. But what is the benefit over creating this value object with a constructor? In my opinion, a builder pattern makes sense if you 

1) have lots of values where a constructor would be confusing because you have to count the parameters positions
2) have defaults and some or all values are optional. 

If you have required values, I would put in some method that checks if all values were set, to protect against programmer error (just in case the unit tests accidentally did not catch this). In this case the method would be called inside each getter.

For the membership validation request all values are required. 

A constructor with arguments would make this class much shorter and convey the required values better.

What do you think?

Reply:
Right, I forgot about forgetting values.

This is the pattern followed in the new UC you added some time ago if I'm not mistaken, will double check later. I thought this is what I had settled on when I was still in the FUN team... but find it hard to imagine I would have fully forgotten about forgetting values. Will also check that soonz.

Reply:
I do agree that the more values you have in the constructor of a ValueObject, the more important to not have a regular constructor. So not as important here as in some places.

What I did here follows the same pattern as `UpdateDonorRequest` and `ValidateDonorRequest`. I'm not sure it is actually worthwhile to add a guard against not putting in all fields in these classes. If you miss one, the tests will catch it anyway.